File: datapack_development/data/scripts/custom/Falk/Falk.py
import sys
from java.lang import System
from com.l2jserver import L2DatabaseFactory
from com.l2jserver.gameserver.model.quest import State
from com.l2jserver.gameserver.model.quest import QuestState
from com.l2jserver.gameserver.model.quest.jython import QuestJython as JQuest
from com.l2jserver.gameserver.instancemanager import HellboundManager

falk = 32297
CaravanCertificates = [9850, 9851, 9852]

class Falk (JQuest):

    def __init__(self, id, name, descr):
        JQuest.__init__(self, id, name, descr)

    def onTalk (self, npc, player):
        st = player.getQuestState("Falk")
        if not st:
            st = self.newQuestState(player)
        have = 0
        for i in CaravanCertificates:
            if st.getQuestItemsCount(i) >= 1:
                have = 1
        if have == 1: return "<html><body>Falk:<br>Did you already give me the Darion's Badges, remember? Now you must deal with somebody other.</body></html>"
        if st.getQuestItemsCount(9674) < 20: return "<html><body>Falk:<br>I am afraid, it is not enough of Darion's Badges. Please, find yet and come back.</body></html>"
        st.takeItems(9674, 20)
        st.giveItems(9850, 1)
        return "<html><body>Jude:<br>Thank you! Take it. Now you will be able to trade with other Caravan's groups.</body></html>"

QUEST = Falk(-1, "Falk", "custom")

QUEST.addStartNpc(falk)
QUEST.addTalkId(falk)

File: datapack_development/data/scripts/hellbound/Falk/Falk.java
package hellbound.Falk;

import com.l2jserver.gameserver.model.actor.L2Npc;
import com.l2jserver.gameserver.model.actor.instance.L2PcInstance;
import com.l2jserver.gameserver.model.quest.Quest;
import com.l2jserver.gameserver.model.quest.QuestState;

/**
 * @author theOne
 */
public class Falk extends Quest
{
	private static final int Falk = 32297;

	private static final int[] CaravanCertificates = {
			9850, 9851, 9852
	};

	public Falk(int id, String name, String descr)
	{
		super(id, name, descr);
		addStartNpc(Falk);
		addTalkId(Falk);
	}

	@Override
	public String onTalk(L2Npc npc, L2PcInstance player)
	{
		QuestState st = player.getQuestState(getName());

		if (st == null)
			st = newQuestState(player);

		boolean haveCertificates = false;
		for (int i : CaravanCertificates)
		{
			if (st.getQuestItemsCount(i) >= 1)
				haveCertificates = true;
		}

		if (haveCertificates)
			return "32297-1.htm";

		if (st.getQuestItemsCount(9674) >= 20)
		{
			st.takeItems(9674, 20);
			st.giveItems(9850, 1);
			return "32297-3.htm";
		}
		else
			return "32297-2.htm";
	}

	public static void main(String[] args)
	{
		new Falk(-1, "Falk", "hellbound");
	}
}